mappymatch/utils/plot.py

```diff
@@ -178,3 +192,91 @@ def plot_map(tmap: NxMap, m=None):
+    """
+
+    # Road Data Frame Section
+    # Define a pandas data frame containing the list of matches (each represented by m) where m.road = True
+
+    # mid_i = int(
+    #     len(coord_df) / 2
+    # )  # find the middle index of the coord_gdf data frame.
+    # mid_coord = coord_df.iloc[
+    #     mid_i
+    # ].geometry  # answer the question: what is the middle coordinate?
+
+    y = coord_df.distance  # the distances from the expected line. Deviance.
+    x = [x for x in range(0, len(y))]  # create blanks for x axis
+
+    # for (
+    #     coord
+    # ) in (
+    #     coord_df.itertuples()
+    # ):  # for every coordinate tuple within coord_gdf ...
+    #     x_coord = coord.geometry.x  # identify the x coordinate geometry.
+    #     y_coord = coord.geometry.y  # identify the y coordinate geometry.
+
+    # for (
+    #     road
+    # ) in road_df.itertuples():  # for every road in the road_gdf data frame...
+    #     full_line = [
+    #         (lat, lon) for lon, lat in road.geometry.coords
+    #     ]  # identify the full line of that road in lat,long tuples.
+
+    # Plotting Section
+    plt.figure(figsize=(15, 7))  # create a figure sized 15 x 7
+    plt.autoscale(
+        enable=True
+    )  # autoscale the figure's contents to the data once it is plotted.
+    # todo ---- what was the purpose of plt.stem? -->  plt.stem(x, y)
+    plt.scatter(
+        x, y
+    )  # create a scatter plot of our x (blanks), and our y (deviance from expected line) values.
+    plt.title("Distance To Nearest Road")  # create a title for our plot.
+    plt.ylabel("Meters")  # establish the y axis label "Meters"
+    plt.xlabel(
+        "Point Along The Path"
+    )  # label the x axis label "Point Along The Path"
+    plt.show()  # print the plot.
+
+
+def plot_prep(file_path):  #
+    """
+    Summary:
+       provided a file path, the plot_prep function creates a trace, geofence, road_map, and matcher using LCSSMatcher and then passes the matches object to the plot_matches function. This function generates a fmap, but it also runs the plot_match_distances function before it finishes executing. With some tweaking this can be customized to the users preferences.
+
+    Args:
+        file_path (str, optional): _description_. Defaults to 'resources/traces/sample_trace_1.csv'.
+    """
+    try:
+        trace = Trace.from_csv(root() / f"{file_path}")
+    except Exception:
+        trace = Trace.from_csv(
+            f"{file_path}"
+        )  # catches any file entry errors.
+    geofence = geofence_from_trace(trace, padding=1e3)
+    road_map = read_osm_nxmap(geofence)
+    matcher = LCSSMatcher(road_map)
+    matches = matcher.match_trace(trace)
+    fmap_result = plot_matches(
+        matches, road_map
+    )  # call the plot_matches function which will plot the matches with matplotlib
+    return fmap_result
+
+
+# change file path to the desired matches as a csv file.
```

Comment:
Do you mind removing these from the file (280-282)? We can probably just test this locally in a notebook or a standalone script.

Reply:
I created a jupyter notebook that lives in the Examples folder that can test the plotting functionality as you requested.